### DISCUSSION TITLE
How to optimize using Docker?

### DISCUSSION THREAD

**Discussion by Harry040** [EA31337/EA31337#330] [Q&A]
Could you explain some hints

**Comment by kenorb** [comment]
As part of [EA-Tester project](https://github.com/EA31337/EA-Tester), you can use the generated container with the following [options](https://github.com/EA31337/EA-Tester/blob/master/scripts/options.txt).

### Backtest

Here is the example of running backtest by specifying a remote EA file:

    docker run ea31337/ea-tester:EURUSD-2020-DS run_backtest -e https://github.com/EA31337/EA31337/releases/download/v2.009/EA31337-Lite-v2.009.ex5 -D5 -v

### Optimize

Similar as above, but for optimization, you you need to add `-o` option and specify the SET file using `-f`.

        docker run ea31337/ea-tester:EURUSD-2020-DS run_backtest -o -f EA_OrderCloseTime.set -e https://github.com/EA31337/EA31337/releases/download/v2.009/EA31337-Advanced-v2.009.ex4 -D5 -v

Notes:

- If you're specifying SET file which exists outside of the Docker container (on your host machine), you need to mount the volume by `-v` param, e.g. `docker run -v localdir:/opt/sets` with the rest of the arguments, so the file exists within the running container.
- For more details, read: [Backtesting using Docker](https://github.com/EA31337/EA31337/wiki/Backtesting-using-Docker).

I'm going to verify and test the above examples, and come back to you with some specific example.
